lua/nvim-lsp-installer/installers/go.lua

```diff
@@ -22,17 +20,20 @@ function M.packages(packages)
                 stdio_sink = ctx.stdio_sink,
             }
 
-            ctx.receipt:with_primary_source(ctx.receipt.go(pkgs[1]))
-            for i = 2, #pkgs do
-                ctx.receipt:with_secondary_source(ctx.receipt.go(pkgs[i]))
-            end
+            ctx.receipt:with_primary_source(ctx.receipt.go(package))
+
+            local pkg, version = unpack(vim.split(package, "@"))
```

Comment:
I think you are correct. I see that my lack of lua knowledge is seeping through and I highly appreciate you taking a good look at this PR. It's valuable for me to learn. :smile:

Reply:
This could be achieved by overriding `ctx.requested_server_version`, like:

```lua
installer = {
    context.set(function (ctx)
        ctx.requested_server_version = "stable" -- always install stable, no matter if user has requested a different version
        ctx.requested_server_version = coalesce(ctx.requested_server_version, "stable") -- default to stable
    end),
    go.package "golang.org/x/tools/gopls"
}
```

I think targeting `@latest` by default (and putting this default inside the `go.package` function`) would make most sense? In cases where defaulting to a different tag than `@latest`, the above snippet could be used instead!

Reply:
Separating the package and the version specifier programmatically like this also makes it easier to do other things. No need to do string parsing to extract them! It also lessens the ambiguity - you don't have to wonder if a string has already been parsed or not when operating on a go package string; is `package` `gopls@latest` or just `gopls`?